ttnn/cpp/ttnn/operations/pool/generic/pool2d_utils.cpp

```diff
@@ -0,0 +1,280 @@
+                                            // of c if its greater than 8 tiles)
+    uint32_t out_cb_npages = output_memory.shard_spec.value().shape[0] * in_ntiles_c;
+    uint32_t out_cb_config_size = out_cb_npages /*108*/ * out_cb_pagesize /*62*/;
+
+    uint32_t max_pool_partials_cb_config_size = 0;
+    if (is_large_kernel) {
+        uint32_t max_pool_partials_cb_pagesize = out_cb_pagesize;
+        uint32_t max_pool_partials_cb_npages = nblocks;
+        max_pool_partials_cb_config_size = max_pool_partials_cb_npages /*1*/ * max_pool_partials_cb_pagesize /*64*/;
+    }
+
+    return in_scalar_cb_config_size
+           // + raw_in_cb_config_size
+           // + in_reader_indices_cb_config_size
+           + in_cb_config_0_size +
+           in_cb_config_1_size
+           // + out_cb_config_size
+           + max_pool_partials_cb_config_size;
+}
+
+sliding_window::ParallelConfig determine_pool_config_for_auto_shard(
+    const Tensor& input_tensor, const sliding_window::SlidingWindowConfig& sliding_window_config, uint32_t channels) {
+    auto batch_size = sliding_window_config.batch_size;
+    auto output_shape = sliding_window_config.get_output_shape();
+    auto compute_grid_size = input_tensor.device()->compute_with_storage_grid_size();
+
+    auto input_parallel_config_height = conv::determine_parallel_config(
+        TensorMemoryLayout::HEIGHT_SHARDED,
+        batch_size,
+        channels,
+        output_shape[1],
+        output_shape[2],
+        channels,
+        compute_grid_size,
+        ShardOrientation::ROW_MAJOR,
+        false,
+        false);
+
+    auto input_parallel_config_width = conv::determine_parallel_config(
+        TensorMemoryLayout::WIDTH_SHARDED,
+        batch_size,
+        channels,
+        output_shape[1],
+        output_shape[2],
+        channels,
+        compute_grid_size,
+        ShardOrientation::ROW_MAJOR,
+        false,
+        false);
+    auto output_parallel_config_width =
+        conv::determine_output_parallel_config(input_parallel_config_width, compute_grid_size, channels, false);
+
+    auto input_parallel_config_block = conv::determine_parallel_config(
+        TensorMemoryLayout::BLOCK_SHARDED,
+        batch_size,
+        channels,
+        output_shape[1],
+        output_shape[2],
+        channels,
+        compute_grid_size,
+        ShardOrientation::COL_MAJOR,
+        false,
+        false);
+
+    auto get_memconfig = [&](const ParallelConfig& parallel_config) {
+        uint32_t nhw = batch_size * output_shape[1] * output_shape[2];
+        uint32_t out_channeal_padded = tt::round_up(
+            channels, conv::get_num_cores_channels_from_parallel_config(parallel_config) * tt::constants::TILE_WIDTH);
+        return conv::create_sharded_memory_config_from_parallel_config(
+            ttnn::Shape({1, 1, nhw, out_channeal_padded}), parallel_config, tt::constants::TILE_HEIGHT);
+    };
+
+    uint32_t l1_usage_height = calculate_L1_usage(
```

Comment:
maybe making a function which figures out l1 usage for a shard spec e2e is worth it in order to not repeat the code